Auth.py

```diff
@@ -299,4 +306,78 @@ def blacklist_token(self, token):
             self.blacklisted_tokens.append(token_list[i]["token"])
 
         conn.commit()
-        conn.close()
\ No newline at end of file
+        conn.close()
+        
+    def sign_up_test(self):
+        #Make sure the user is already deleted
+        conn = pymysql.connect(self.host, self.username, self.password, self.database, cursorclass=pymysql.cursors.DictCursor)
+        cursor = conn.cursor()
+
+        cursor.execute("DELETE FROM users WHERE (firstname = 'Test' AND lastname = 'Test' );")
+
+        conn.commit()
+        conn.close()
+            
+        error_messages = ["empty-first-name", "empty-last-name", "empty-email", 
+                              "email-already-taken", "invalid-email", 
+                              "password-too-short", "passwords-no-match"]
+        result = self.sign_up( firstname="Test", lastname="Test", email="test@gmail.com", password="longpassword", confirmpassword="longpassword")
+
+        if result in error_messages:
+            print("Something went wrong in the sign_up_test, the error was: " + result)
+        else:
+            print("Everything went fine in the sign_up_test")
+        #These functions run the individual tests
+        
+
+    def sign_in_test(self):
+        error_messages = ["user-no-exist", "incorrect-email-or-password"]
+
+        result = self.sign_in( email="test@gmail.com", password="longpassword")
+
+        if result in error_messages:
+            print("Something went wrong in the sign_in_test, the error was: " + result)
+        else: 
+            print("Everything went fine in the sign_in_test")
+        
+
+    def validate_token_test(self):
+        error_message = ["ERROR_INVALID_TOKEN"]
+
+        result = self.validate_token( token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJoaWdobG93IiwiaWF0IjoxNTU2MTUxMzI0LjAsInN1YiI6IjM3ZTIxZjI2LTY2ZWYtMTFlOS04YTliLTAyNDJhYzExMDAwMiIsImV4cCI6MTU3MTkxOTMyNC4wfQ.s2payesaWIfMMNwF0ofNHsxSyGACFRVhj3gh4C8AQp8")
+
+        if len( result.split(".") ) == 3:
+            if result not in error_message:
+                print("Everything went fine in the validate_token_test")  
+            else:
+                print("Something went wrong in the validate_token_test, the error was: " + result)    
+        
+
+    def send_password_reset_email_test(self):
+        result = self.send_password_reset_email( email="test@gmail.com")
+
+        if result == "success":
+            print("send_password_reset_email was a success")
+        else:
+            print("send_password_reset_email was not a success, the error is: " + result)
+        
+
+    def reset_password_test(self):
+        result = self.reset_password( token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJoaWdobG93IiwiaWF0IjoxNTU2MTUxMzI0LjAsInN1YiI6IjM3ZTIxZjI2LTY2ZWYtMTFlOS04YTliLTAyNDJhYzExMDAwMiIsImV4cCI6MTU3MTkxOTMyNC4wfQ.s2payesaWIfMMNwF0ofNHsxSyGACFRVhj3gh4C8AQp8",
```

Comment:
Again, use `sign_in` to get an always-valid token:
```suggestion
        token = self.sign_in("test@example.com", "longpassword")
        result = self.reset_password( token , "longpassword", "longpassword")
```

Reply:
GitHub won't let me select multiple lines to make changes to, so make sure to delete the second line that specifies the password and confirm password.